### PR TITLE
feat: [CDS-50702]: Fixed issues for terragrunt provisioners

### PR DESCRIPTION
**Pull request by sonaligoyal-harness**
### Summary

1. Retained Workspace value for Terragrunt Destroy and apply step in UI
2. Fixed Submit button for BackendConfig in Terragrunt apply and destroy 
3. Updated terragruntRunType for Terragrunt plan step
4. Supported runtime for path field in terragrunt plan step
5. Added allowed values support for fields

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/75-cd/components/PipelineSteps/TerragruntPlan/InputSteps/TgPlanInputStep.tsx [hunk 1]
```diff
@@ -102,6 +102,25 @@ function TgPlanInputStep(props: TerragruntPlanProps & { formik?: FormikContextTy
...
+            name={`${pathPrefix}spec.configuration.moduleConfig.path`}
+            disabled={readonly}
+            template={inputSetData?.template}
+            fieldPath={'spec.configuration.moduleConfig.path'}
+            multiTextInputProps={{
+              expressions,
+              allowableTypes
+            }}
+          />
+        </div>
```
Comment (rameshx) [inline src/modules/75-cd/components/PipelineSteps/TerragruntPlan/InputSteps/TgPlanInputStep.tsx:121]: `TextFieldInputSetView` has an internal grid so wrapping div is not required. You can just pass the classNames.
```suggestion
        <TextFieldInputSetView
          placeholder={'Enter path'}
          label={getString('common.path')}
          name={`${pathPrefix}spec.configuration.moduleConfig.path`}
          disabled={readonly}
          template={inputSetData?.template}
          fieldPath={'spec.configuration.moduleConfig.path'}
          multiTextInputProps={{
            expressions,
            allowableTypes
          }}
          className={cx(stepCss.formGroup, stepCss.md)}
        />
```

File: src/modules/75-cd/components/PipelineSteps/Common/Terragrunt/EditView/TerragruntEditView.tsx [hunk 2]
```diff
@@ -693,7 +694,7 @@ export default function TerragruntEitView(
                                   {backendConfigFilePath && (
                                     <>
                                       <Text font="normal" lineClamp={1} width={200}>
-                                        {backendConfigFilePath}
+                                        /{backendConfigFilePath}
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/components/PipelineSteps/Common/Terragrunt/EditView/TerragruntEditView.tsx:697]: Is `/` intentional. If yes, use 
```suggestion
                                        {`/${backendConfigFilePath}`}
```

File: src/modules/75-cd/components/PipelineSteps/TerragruntPlan/InputSteps/TgPlanInputStep.tsx [hunk 3]
```diff
@@ -102,6 +102,25 @@ function TgPlanInputStep(props: TerragruntPlanProps & { formik?: FormikContextTy
           />
         </div>
       )}
+
+      {getMultiTypeFromValue(get(inputSetData?.template, 'spec.configuration.moduleConfig.path')) ===
+        MultiTypeInputType.RUNTIME && (
+        <div className={cx(stepCss.formGroup, stepCss.md)}>
+          <TextFieldInputSetView
+            placeholder={'Enter path'}
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/components/PipelineSteps/TerragruntPlan/InputSteps/TgPlanInputStep.tsx:110]: This should come from yaml. Avoid hard coding label.

File: src/modules/75-cd/components/PipelineSteps/Common/Terragrunt/InputSteps/TerragruntInputStep.tsx [hunk 4]
```diff
@@ -148,8 +150,26 @@ export default function TerragruntInputStep<T extends TerragruntData = Terragrun
         return <></>
       })}
 
-      {getMultiTypeFromValue(get(inputSetData?.template, 'spec.configuration.spec.backendConfig.spec.content')) ===
-        MultiTypeInputType.RUNTIME && (
+      {getMultiTypeFromValue(config?.spec?.workspace) === MultiTypeInputType.RUNTIME && (
```
Comment (AshwinBhatkal-Harness) [inline src/modules/75-cd/components/PipelineSteps/Common/Terragrunt/InputSteps/TerragruntInputStep.tsx:153]: Better to use isValueRuntimeInput util to check this. Will make it easier with execution inputs

File: src/modules/75-cd/components/PipelineSteps/TerragruntPlan/InputSteps/TgPlanInputStep.tsx [hunk 5]
```diff
@@ -102,6 +102,24 @@ function TgPlanInputStep(props: TerragruntPlanProps & { formik?: FormikContextTy
           />
         </div>
       )}
+
+      {getMultiTypeFromValue(get(inputSetData?.template, 'spec.configuration.moduleConfig.path')) ===
```
Comment (AshwinBhatkal-Harness) [inline src/modules/75-cd/components/PipelineSteps/TerragruntPlan/InputSteps/TgPlanInputStep.tsx:106]: same here

**Review by AshwinBhatkal-Harness — DISMISSED** [review]

File: src/modules/75-cd/components/PipelineSteps/Common/Terragrunt/InputSteps/TerragruntInputStep.tsx (hunk 4 above)
Comment (sonaligoyal-harness) [reply src/modules/75-cd/components/PipelineSteps/Common/Terragrunt/InputSteps/TerragruntInputStep.tsx:153]: Sure, will update it.

**Review by rameshx — APPROVED** [review]

**Review by Sagar-Rajput7 — APPROVED** [review]

**Pull request merged**